strutil/pathiter.go

```diff
@@ -58,48 +61,94 @@ func NewPathIterator(path string) (*PathIterator, error) {
 	return &PathIterator{path: path}, nil
 }
 
-// Path returns the path being traversed.
+// Path returns the entire path being traversed.
 func (iter *PathIterator) Path() string {
 	return iter.path
 }
 
-// CurrentName returns the name of the current path element.
-// The return value may end with '/'. Use CurrentNameNoSlash to avoid that.
-func (iter *PathIterator) CurrentName() string {
-	return iter.path[iter.left:iter.right]
-}
-
-// CurrentNameNoSlash returns the same value as Name with right slash trimmed.
-func (iter *PathIterator) CurrentNameNoSlash() string {
-	if iter.right > 0 && iter.path[iter.right-1:iter.right] == "/" {
+// CurrentBase returns the last element of the current path.
+// It removes any trailing slash unless the last element of the
+// current path is the filesystem root.
+// Before any iteration with Next, CurrentBase returns an empty string.
+// Example:
+//
+//	iter:= NewPathIterator("/foo/bar")  // iter.CurrentBase() == ""
+//	iter.Next() // iter.CurrentBase() == "/" && iter.CurrentPath() == "/"
+//	iter.Next() // iter.CurrentBase() == "foo" && iter.CurrentPath() == "/foo"
+//	iter.Next() // iter.CurrentBase() == "bar" && iter.CurrentPath() == "/foo/bar"
+func (iter *PathIterator) CurrentBase() string {
+	// Only remove trailing slash if the iterator is not pointing to the filesystem root
+	if iter.right > 1 && iter.path[iter.right-1:iter.right] == "/" {
 		return iter.path[iter.left : iter.right-1]
 	}
 	return iter.path[iter.left:iter.right]
 }
 
-// CurrentPath returns the prefix of path that was traversed, including the current name.
+// CurrentPath returns the path up to the current depth.
+// Before any iteration with Next, CurrentPath returns an empty string.
+// Example:
+//
+//		iter:= NewPathIterator("/foo/bar")  // iter.CurrentPath() == ""
```

Comment:
```suggestion
//		iter := NewPathIterator("/foo/bar")  // iter.CurrentPath() == ""
```

This applies to some other lines below.